rfcs/0080-nixos-release-schedule

```diff
@@ -0,0 +1,69 @@
+---
+feature: nixos-release-schedule
+start-date: 2020-11-28
+author: Jonathan Ringer (@jonringer)
+co-authors: Frederik Rietdijk (@FRidh)
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+NixOS is released twice a year; the current release schedule targets March
+and September for the release months. However, trying to provide a polished
+desktop-user experience is difficult and moving the target release months to
+May and November should minimize the amount of in-house work needed to
+cut a release.
+
+# Motivation
+[motivation]: #motivation
+
+By changing our release dates to follow those of certain upstream projects,
+we could reduce the amount of issues we encounter during stabilization.
+In particular, GNOME and KDE Plasma both have a release in October, which
```

Comment:
GNOME 3.38 released on [September 16](https://wiki.gnome.org/Forty), we only merged it on [October 24](https://github.com/NixOS/nixpkgs/pull/98316#event-3916100085) and that included only rough testing by the GNOME packagers team (all core apps start and we did not experience any serious issues when running GNOME session). It was merged to master on [November 10](https://github.com/NixOS/nixpkgs/pull/102739#event-3977835342), which allowed wider user base to start testing this. That revealed a [serious regression](https://github.com/NixOS/nixpkgs/issues/103746) that GNOME maintainers were not able to reproduce as of yet. With this proposal a branch-off would be somewhere around the end of October so I doubt the delay will be enough to produce a well tested and stable release.

Reply:
historically, the branch-off was done in the middle of the month, so 20.03 I think was around the ~14th, for 20.09 it was the beginning of the second second week, so I think around the 7th. And the target release date is usually on the last week of the month.

I have another RFC proposal (which I am in the process of creating) which would create a schedule for doing the desktop manager tasks before ZHF. Remember that the majority of the work for GNOME 3.38 was done while ZHF was still going on, and efforts were divided. If we have a few weeks were the community is focused on packaging plasma and gnome, I don't think the time to package them will be so long.

Also, organizing the effort with materials for people to "onboard" onto the effort helps a lot, something like https://github.com/NixOS/nixpkgs/issues/101964 where I wrote up  how to go about doing the fixes, and then tracked the overall progress of the effort.

Reply:
I fixed the release month from October to September (not sure why I put October).

Reply:
@jtojnar Would it perhaps be possible to start the GNOME packages update a few weeks before release based on the (at that time) current development versions?

Would that make the bump of the actual release small enough to allow for a tighter window?

Reply:
Last few cycles we actually started packaging beta or alpha releases. The updates themselves are mostly automated so the biggest time sink is going through all the changelogs. We eventually want to have a nixpkgs branch tracking git `HEAD`s of all the GNOME packages but need a tooling first – with unreleased code, we do not have the benefit of curated changelogs.